### PR TITLE
Log the command line flags

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

With d7ddcca231612c618dce4e9895478167d397cbd7, we lost the logging
of the flags. We should at least log what the command line flags
were used to start processes as those incredibly useful for trouble shooting.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/assign @deads2k 
/assign @liggitt 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/util/flag/flags.go [hunk 1]
```diff
@@ -43,12 +45,16 @@ func WarnWordSepNormalizeFunc(f *pflag.FlagSet, name string) pflag.NormalizedNam
...
 	pflag.Parse()
-	pflag.VisitAll(func(flag *pflag.Flag) {
-		glog.V(2).Infof("FLAG: --%s=%q", flag.Name, flag.Value)
-	})
 }
+
+// PrintFlags logs the command line parameters
+func PrintFlags() {
+	for _, arg := range os.Args {
+		log.Printf("FLAG: %v", arg)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: why not glog.V(2).Infof?

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: also, os.Args is not the same as flags... does this mean `kubelet -h` would print this?
```
FLAG: kubelet
FLAG: -h
```

Comment (dims) [reply staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: `log.Printf` was from trial and error. `glog.V(2).Infof` does not work at this point.

let me check on the 'kubectl -h`

File: cmd/kubelet/app/server.go [hunk 2]
```diff
@@ -155,9 +154,6 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 			// short-circuit on verflag
 			verflag.PrintAndExitIfRequested()
 
-			// log args (separate lines, so we don't overflow line-length limits in logging infrastructure)
-			glog.V(2).Infof("kubelet flags: %s", strings.Join(args, "\n"))
```
Comment (mtaufen) [inline cmd/kubelet/app/server.go]: Kubelet intentionally avoids using the global flagset, and as a result the flag logging should be done here. As @liggitt said below, `os.Args` is not the same as flags.

Comment (dims) [reply cmd/kubelet/app/server.go]: ack. i'll add this back

File: staging/src/k8s.io/apiserver/pkg/util/flag/flags.go (hunk 1 above)
Comment (dims) [reply staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: hmm, you meant kubelet not kubectl

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: yes, sorry

**Comment by dims** [comment]
@mtaufen @liggitt - found a better spot to log the flags, seems to work better. please take a look. `kubelet -h` or `kubectl -h` should not print the flags for sure with this change.

File: staging/src/k8s.io/apiserver/pkg/util/flag/flags.go [hunk 3]
```diff
@@ -43,12 +43,16 @@ func WarnWordSepNormalizeFunc(f *pflag.FlagSet, name string) pflag.NormalizedNam
 	return pflag.NormalizedName(name)
 }
 
-// InitFlags normalizes, parses, then logs the command line flags
+// InitFlags normalizes, parses the command line flags
```
Comment (mtaufen) [inline staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: nit: `normalizes and parses`

File: staging/src/k8s.io/apiserver/pkg/util/flag/flags.go [hunk 4]
```diff
@@ -43,12 +43,16 @@ func WarnWordSepNormalizeFunc(f *pflag.FlagSet, name string) pflag.NormalizedNam
...
 func InitFlags() {
 	pflag.CommandLine.SetNormalizeFunc(WordSepNormalizeFunc)
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
 	pflag.Parse()
-	pflag.VisitAll(func(flag *pflag.Flag) {
+}
+
+// PrintFlags logs the command line parameters
```
Comment (mtaufen) [inline staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: nit: s/parameters/flags

**Review by mtaufen — COMMENTED** [review]
couple nits otherwise LGTM

File: staging/src/k8s.io/apiserver/pkg/util/flag/flags.go (hunk 3 above)
Comment (dims) [reply staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: fixed

File: staging/src/k8s.io/apiserver/pkg/util/flag/flags.go (hunk 4 above)
Comment (dims) [reply staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: fixed

File: staging/src/k8s.io/apiserver/pkg/util/flag/flags.go [hunk 5]
```diff
@@ -43,12 +43,16 @@ func WarnWordSepNormalizeFunc(f *pflag.FlagSet, name string) pflag.NormalizedNam
...
+// InitFlags normalizes and parses the command line flags
 func InitFlags() {
 	pflag.CommandLine.SetNormalizeFunc(WordSepNormalizeFunc)
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
 	pflag.Parse()
-	pflag.VisitAll(func(flag *pflag.Flag) {
+}
+
+// PrintFlags logs the flags in the flagset
+func PrintFlags(flags *pflag.FlagSet) {
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: I'd rather have this here and duplicated in k8s.io/kubernetes (near the verflag probably) than have everything depend on this from apiserver.

I'd make it V(1).

**Comment by deads2k** [comment]
one comment on the structure/deps.  lgtm otherwise

/approve

**Comment by deads2k** [comment]
@mikedanese you're a cmd approver, ptal.

File: staging/src/k8s.io/apiserver/pkg/util/flag/flags.go (hunk 5 above)
Comment (dims) [reply staging/src/k8s.io/apiserver/pkg/util/flag/flags.go]: thanks @deads2k will update the PR shortly

**Comment by dims** [comment]
@deads2k i've fixed the nits, can you please re-approve?

File: pkg/util/flag/flags.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+package flag
+
+import (
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+// PrintFlags logs the flags in the flagset
+func PrintFlags(flags *pflag.FlagSet) {
```
Comment (lavalamp) [inline pkg/util/flag/flags.go]: Please avoid adding a dependency to glog by accepting a `logLine func(format string, args... interface{})` parameter.

Comment (lavalamp) [reply pkg/util/flag/flags.go]: Hm, I guess this is a completely new package? I'm surprised we don't have a place to put this.

**Comment by lavalamp** [comment]
If you like this behavior, please write a test.

/approve

My comment is optional.

**Comment by dims** [comment]
@briantopping sure thing! feel free to file a new pr. i'll be happy to `lgtm` it.

**Pull request merged**